### PR TITLE
RSC rifle crate as cargo purchase

### PR DESCRIPTION
**Pull request by Unisol**
## About the PR
Added RSC Model 1917, a Nanotrasen-approved Great War rifle. Available for purchase for supply department. Packed by senile veteran of six stellar wars who can't quite remember how many rifles and stripper clips should go in one crate; your mileage will vary.

## Why / Balance
Syndicate has cheap mass infantry option, and now Nanotrasen does too. Meant to be a cheap way to arm large amounts of people with questionable quality guns. RSC 1917 is mechanically identical to Mosin Nagant.

Crates are can be unlocked using general security or cargo access.

Cost: one crate goes for 2000$, with one rifle costing $400 and one clip costing $5. Anywhere between 2 and 5 rifles per crate, with anywhere between 4 and 42 clips mean that total cost varies between $1020 and $2210. This ensures cargo can't reliably resell these at a profit.

Stripper clips in secfab go for one steel sheet per one clip with five rounds. This is in line with other .30 ammo options, which come out to 0.18 steel per round.

## Media

cargo console
crate closed
crate open
SS14 Loader_p2ye9SpGqD
SS14 Loader_HzVz1vMQCp
SS14 Loader_DWzTE8xW32
SS14 Loader_fpNUsGRQO4

## Requirements

- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

**Changelog**

:cl:
- add: added RSC Model 1917 for purchase in supply dept
- add: added stripper clips with .30 ammo to security and ammo techfabs, as well as emagged protolathe

### CODE REVIEW

**Review by DieselMohawk — APPROVED** [review]
Good sprites.

I have no idea if we should reference real world locations and events with this rifle, even the name is a bit too historically accurate for the SS14 vibe, if you get what I mean.

Anyways, good sprites.

**Comment by FluffMe** [comment]
@Unisol test fail is real

`TestSufficientSpaceForEntityStorageFill` fails with `CrateArmoryRSC storage fill is too large.`

File: Resources/Prototypes/_Harmony/Catalog/Fills/Crates/armory.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      - id: WeaponSniperRSC
+        orGroup: guns
+        amount: 3
+      - id: WeaponSniperRSC
+        orGroup: guns
+        amount: 4
+      - id: WeaponSniperRSC
+        orGroup: guns
+        amount: 5
+        # Lost is a dogshit show
```
Comment (FluffMe) [inline Resources/Prototypes/_Harmony/Catalog/Fills/Crates/armory.yml:23]: image

**Review by FluffMe — CHANGES_REQUESTED** [review]
- Test fail should be addressed (if amount of rifles/ammo changes significantly price may need to be reduced).
- Lost comment, while appreciated and agreed with, should be removed.
- Cargo access on the crate will finally prevent slamming it with pickaxe like idiots. But it differs from other gun purchases. If that's an intended change, I agree with this, but please test if it overwrites `access: [["Armory"]]` of `CrateWeaponSecure` and if Armory access should be added (maybe instead of security even, who knows. No idea how it works without testing).

Thanks for this addition. Hail Cargonia!
I'll read and test this more thoroughly when changes are addressed.

**Comment by FluffMe** [comment]
> I have no idea if we should reference real world locations and events with this rifle, even the name is a bit too historically accurate for the SS14 vibe, if you get what I mean.

There is a precedent with Mosin, and they have the same vibe IMO.
We also already discussed if the gun is in public domain or not, so this is also not a concern.

**Comment by FluffMe** [comment]
PS: I especially dig the RNG factor of the crate XD

**Comment by Unisol** [comment]
Increased CrateArmoryRSC capacity to 50; removed truthful comment; added Armory access to the crate in addition to Supply and Securty.

**Comment by FluffMe** [comment]
Another test fail: `NoStaticPriceAndStackPrice`

**Comment by Unisol** [comment]
No idea what this is about, I've checked crate, rifle and clip - all have only static price and not stack price.

**Comment by FluffMe** [comment]
I'll debug when I get some time next week.

**Comment by FluffMe** [comment]
Was a Heisentest XD

**Review by DieselMohawk — APPROVED** [review]

**Review by FluffMe — COMMENTED** [review]
Code is good! great job, thank you.

I'll leave it hanging for a bit for admins and Keld to pitch in, since it's a new weapon and weapon source for the crew.

**Comment by liltenhead** [comment]
Considering there's an AK and a Mosin in game, I like the idea. Sprite is great looking too.

**Review by FluffMe — APPROVED** [review]

**Pull request merged**